### PR TITLE
fix(route): 修复 UP 主动态 路由

### LINKED ISSUE #14990: Bilibili UP主动态路由失效
### 路由地址

```routes
/bilibili/user/dynamic/:uid/:routeParams?
```

### 完整路由地址

```fullroutes
/bilibili/user/dynamic/199676483
```

### 相关文档

https://docs.rsshub.app/zh/routes/social-media#up-%E4%B8%BB%E5%8A%A8%E6%80%81

### 预期是什么？

正常返回

### 实际发生了什么？

Cannot read properties of null (reading 'map')

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

_No response_

### 额外信息

```shell
Cannot read properties of null (reading 'map')
```

### 这不是重复的 issue

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by CaoMeiYouRen**
## Involved Issue / 该 PR 相关 Issue

Close #14990 

## Example for the Proposed Route(s) / 路由地址示例

```routes
/bilibili/user/dynamic/199676483
/bilibili/user/dynamic/2267573
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
添加了 Cookie 配置以应对请求风控

### CODE REVIEW

File: lib/routes/bilibili/dynamic.ts [hunk 1]
```diff
@@ -115,11 +116,16 @@ async function handler(ctx) {
     const useAvid = fallback(undefined, queryToBoolean(routeParams.useAvid), false);
     const directLink = fallback(undefined, queryToBoolean(routeParams.directLink), false);
 
+    const cookie = Object.values(config.bilibili.cookies).find(Boolean); // 寻找一个真值的 cookie，本路由中 cookie 可以是任意账号的
+    if (!cookie) {
+        throw new Error('缺少 Bilibili 用户登录后的 Cookie 值');
```
Comment (TonyRL) [inline lib/routes/bilibili/dynamic.ts:121]: Cookie is not compulsory.

Comment (CaoMeiYouRen) [reply lib/routes/bilibili/dynamic.ts:121]: > Cookie is not compulsory.

At present, if cookies are not set, they will be subject to risk control. So cookies are necessary

**Comment by DIYgod** [comment]
The API currently used on the web side seems to require no login to access it

**Comment by CaoMeiYouRen** [comment]
> The API currently used on the web side seems to require no login to access it

这个我已经试过了，是会生成cookie的，但是在获取这个cookie的时候就有可能会触发风控。
我因为要研究这个接口多试了几次就触发了风控。
目前来看改为v1接口+cookie是最简单的解决方案

**Pull request closed without merging**